### PR TITLE
Detect moving distance for entering Text edit mode.

### LINKED ISSUE #729: Text edit mode often conflicts with dragging operation
## Bug Description

Try to randomly drag text elements around. Especially short moves. Regularly, instead of completing the dragging behavior, the element goes into the edit mode. In such cases it often shows a misaligned frame:

## Expected Behaviour

That should practically never happen.

## Steps to Reproduce

Do short drag operations with a text element: short moves and quick delay between grab and release.

## Screenshots

Broken state:
image

Screencast:

capture

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by miina**
Fixes #729

### CODE REVIEW

File: assets/src/edit-story/elements/text/frame.js [hunk 1]
```diff
@@ -93,15 +94,26 @@ function TextFrame({ element: { id, content, ...rest }, wrapperRef }) {
...
+        return;
+      }
 
-      if (timingDifference > 150) {
-        // Only short clicks count.
+      const distanceMoved = Math.max(
+        Math.abs(evt.clientX - clickCoordinates.x),
+        Math.abs(evt.clientY - clickCoordinates.y)
+      );
+      if (timingDifference > 300 || distanceMoved > 1) {
```
Comment (miina) [inline assets/src/edit-story/elements/text/frame.js]: @dvoytenko This is now restoring the original `300` for click length and additionally checks the distance moved, allowing movement of up to 1 pixel for still entering the edit mode.

In my testing, the value when doing a click was actually always `0`, it needed some effort to do a click which moves meanwhile without the intention of dragging, however, perhaps that's also dependent on the mouse/touchpad.

LMK if you think we should set it to `0` (or assign some other value).

**Review by barklund — APPROVED** [review]
Works for me!

File: assets/src/edit-story/elements/text/frame.js (hunk 1 above)
Comment (dvoytenko) [reply assets/src/edit-story/elements/text/frame.js]: @miina what you have makes a lot of sense to me. Just a couple of nits:

1. The `distanceMoved` should be manhattan distance. E.g. `abs(clientX-.x) + abs(clientY-.y)`.
2. I think threshold can be made bigger to `> 4px` manhattan.

**Review by dvoytenko — CHANGES_REQUESTED** [review]

File: assets/src/edit-story/elements/text/frame.js (hunk 1 above)
Comment (miina) [reply assets/src/edit-story/elements/text/frame.js]: @dvoytenko: The allowed distance is adjusted now.

**Review by dvoytenko — APPROVED** [review]

**Pull request merged**